src/_pytest/pytester.py

```diff
@@ -1065,15 +1082,24 @@ def run(self, *cmdargs, **kwargs):
         :param args: the sequence of arguments to pass to `subprocess.Popen()`
         :param timeout: the period in seconds after which to timeout and raise
             :py:class:`Testdir.TimeoutExpired`
+        :param stdin: optional standard input.  Bytes are being send, closing
+            the pipe, otherwise it is passed through to ``popen``.
+            Defaults to ``CLOSE_STDIN``, which translates to using a pipe
+            (``subprocess.PIPE``) that gets closed.
 
         Returns a :py:class:`RunResult`.
 
         """
         __tracebackhide__ = True
 
         timeout = kwargs.pop("timeout", None)
+        stdin = kwargs.pop("stdin", CLOSE_STDIN)
         raise_on_kwargs(kwargs)
 
+        popen_kwargs = {"stdin": stdin}
+        if isinstance(stdin, bytes):
```

Comment:
It doesn't seem like you need to handle this explicitly, you may pass `stdin` directly through `self.popen` and it will do the right thing.

Reply:
Yes, only closing is needed then.  Thanks!